[ZEPPELIN-5191] Fix the wrong Docker image name in zeppelin-server.yaml

maziyarpanahi opened the pull request:
- There has never been any Docker image named `zeppelin-interpreter` or `zeppelin-server`.
- The latest version also should be 0.10.1

### What is this PR for?
A few sentences describing the overall goals of the pull request's commits.
First time? Check out the contributing guide - https://zeppelin.apache.org/contribution/contributions.html

All the examples inside and outside Zeppelin website/docs point to simply using `zeppelin-server.yaml` for K8s cluster : 
```sh
curl -s -O https://raw.githubusercontent.com/apache/zeppelin/master/k8s/zeppelin-server.yaml
kubectl apply -f zeppelin-server.yaml
```

There are lots of issues on StackOverFlow and 1 on Jira that indicate the Docker images mentioned in this YAML simply don't exist on Docker Hub. 

### What type of PR is it?
Documentation
*Please leave your type of PR only*

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/ZEPPELIN/
* Put link here, and add [ZEPPELIN-*Jira number*] in PR title, eg. [ZEPPELIN-533]
https://issues.apache.org/jira/browse/ZEPPELIN-5191

### How should this be tested?
* Strongly recommended: add automated unit tests for any new or changed behavior
* Outline any manual steps to test the PR here.

I have tested this on a live K8s cluster (OpenStack) and it created a Zeppelin server successfully, also creates Spark and other interpreters in the cluster upon request without any issue:

```
curl -s -O https://raw.githubusercontent.com/apache/zeppelin/master/k8s/zeppelin-server.yaml
```

### Screenshots (if appropriate)

### Questions:
* Does the licenses files need to update?
* Is there breaking changes for older versions?
* Does this needs documentation?

jongyoul commented:
@maziyarpanahi Hello, BTW what is this?
```
curl -s -O https://raw.githubusercontent.com/apache/zeppelin/master/k8s/zeppelin-server.yaml
```

maziyarpanahi commented:
> @maziyarpanahi Hello, BTW what is this?
> 
> ```
> curl -s -O https://raw.githubusercontent.com/apache/zeppelin/master/k8s/zeppelin-server.yaml
> ```

Apologies, I forgot the next line:

```
curl -s -O https://raw.githubusercontent.com/apache/zeppelin/master/k8s/zeppelin-server.yaml
kubectl apply -f zeppelin-server.yaml
```

I wanted to make sure everything in this quick start works as expected: https://zeppelin.apache.org/docs/0.10.1/quickstart/kubernetes.html#quickstart

Reamer commented:
Hello @maziyarpanahi

Zeppelin is quite complex and I agree that the documentation is not very useful.

On the whole, I would like to get rid of the apache/zeppelin docker image as it is not really suitable for K8s.

I envision the following process:
Automatically build the [zeppelin-distribution image](https://github.com/apache/zeppelin/blob/master/Dockerfile).
Then build both the [zeppelin-server image](https://github.com/apache/zeppelin/blob/master/scripts/docker/zeppelin-server/Dockerfile) and the [zeppelin-interpreter image](https://github.com/apache/zeppelin/blob/master/scripts/docker/zeppelin-interpreter/Dockerfile) on top of it.

Can you perhaps perform the process as an outsider and adjust the documentation accordingly.

maziyarpanahi commented:
> Hello @maziyarpanahi
> 
> Zeppelin is quite complex and I agree that the documentation is not very useful.
> 
> On the whole, I would like to get rid of the apache/zeppelin docker image as it is not really suitable for K8s.
> 
> I envision the following process: Automatically build the [zeppelin-distribution image](https://github.com/apache/zeppelin/blob/master/Dockerfile). Then build both the [zeppelin-server image](https://github.com/apache/zeppelin/blob/master/scripts/docker/zeppelin-server/Dockerfile) and the [zeppelin-interpreter image](https://github.com/apache/zeppelin/blob/master/scripts/docker/zeppelin-interpreter/Dockerfile) on top of it.
> 
> Can you perhaps perform the process as an outsider and adjust the documentation accordingly.

Hi @Reamer 

Yes, I will follow these steps to 
- build custom images for `server` and `interpreter` based on `zeppelin-distribution image`
- download the default `zeppelin-server.yaml`
- change the images for the server and interpreter to point to these custom local images

I'll share the process here before we decide how to update the docs to reflect these steps. (it should be straightforward)

maziyarpanahi commented:
Hi @Reamer 

I have added the steps to build `zeppelin-distribution`, `zeppelin-server`, and `zeppelin-interpreter` to be set inside `./k8s/zeppelin-server.yaml` script.

Please let me know if you want me to adjust something further.

Reamer commented:
Very good, that is how I imagined it.

I would remove the change in `k8s/zeppelin-server.yaml` and put a reminder in the documentation to adjust the images in the YAML-File.

maziyarpanahi commented:
> Very good, that is how I imagined it.
> 
> I would remove the change in `k8s/zeppelin-server.yaml` and put a reminder in the documentation to adjust the images in the YAML-File.

Thanks @Reamer 
I have removed the parts showing how to remove and replaced it with a simple reminder to adjust the images' names
